cql3/statements/describe_statement.cc

```diff
@@ -135,23 +148,99 @@ future<std::vector<description>> generate_descriptions(
     co_return descs;
 }
 
+template <std::ranges::range Range>
+    requires std::is_base_of_v<keyspace_element, std::ranges::range_value_t<Range>>
+future<std::vector<description>> generate_descriptions(replica::database& db, const Range& elements,
+    std::optional<bool> with_internals = std::nullopt, bool sort_by_name = true)
+{
+    std::vector<description> descs{};
+
+    // In general, a range is allowed to be modified during the call to `std::ranges::size()`.
+    // However, we rely on the sanity of the passed types and the fact that we take the range
+    // as a const-reference.
+    descs.reserve(std::ranges::size(elements));
+
+    for (auto& e: elements) {
+        auto desc = (with_internals.has_value())
+            ? description(db, e, *with_internals)
+            : description(db, e);
+        descs.push_back(std::move(desc));
+
+        co_await coroutine::maybe_yield();
+    }
+
+    if (sort_by_name) {
+        std::ranges::sort(descs, [] (const auto& lhs, const auto& rhs) {
+            return lhs._name < rhs._name;
+        });
+    }
+
+    co_return descs;
+}
+
+template <data_dictionary::keyspace_element_generator KEG>
```

Comment:
maybe the same thing can be achieved simpler than templating? perhaps abstract class?

I am not sure exactly what this approach is trying to solve but seeing stuff like: `   const sstring element_type = KEG::element_type(db);` it's a bit hard to read.